File: Rosella/src/main/java/me/hydos/rosella/device/QueueFamilyIndices.java
package me.hydos.rosella.device;

import java.util.Objects;

public class QueueFamilyIndices {

    public Integer graphicsFamily;
    public Integer presentFamily;

    public boolean isComplete() {
        return graphicsFamily != null && presentFamily != null;
    }

    public int[] unique() {
        if (Objects.equals(graphicsFamily, presentFamily)) {
            return new int[]{graphicsFamily};
        } else {
            return new int[]{graphicsFamily, presentFamily};
        }
    }
}

File: Rosella/src/main/kotlin/me/hydos/rosella/render/device/QueueFamilyIndices.kt
package me.hydos.rosella.render.device

import java.util.stream.IntStream

class QueueFamilyIndices {
	internal var graphicsFamily: Int? = null
	internal var presentFamily: Int? = null

	val isComplete: Boolean
		get() = graphicsFamily != null && presentFamily != null

	fun unique(): IntArray {
		return IntStream.of(graphicsFamily!!, presentFamily!!).distinct().toArray()
	}

	fun array(): IntArray {
		return intArrayOf(graphicsFamily!!, presentFamily!!)
	}
}
